lib/node_modules/@stdlib/math/base/special/besselj0/src/main.c

```diff
@@ -0,0 +1,274 @@
+		ax = -x;
+	} else {
+		ax = x;
+	}
+	if ( ax <= 1.0 ) {
+		s1 = 22779.090197304686 + (x * (41345.38663958076 + (x * (21170.523380864943 + (x * (3480.648644324927 + (x * (153.76201909008356 + (x * 0.8896154842421046)))))))));
+		s2 = 22779.090197304686 + (x * (41370.41249551042 + (x * (21215.350561880117 + (x * (3502.8735138235606 + (x * (157.11159858080893 + (x * 1.0)))))))));
+	} else {
+		ix = 1.0 / x;
+		s1 = 0.8896154842421046 + (ix * (153.76201909008356 + (ix * (3480.648644324927 + (ix * (21170.523380864943 + (ix * (41345.38663958076 + (ix * 22779.090197304686)))))))));
+		s2 = 1.0 + (ix * (157.11159858080893 + (ix * (3502.8735138235606 + (ix * (21215.350561880117 + (ix * (41370.41249551042 + (ix * 22779.090197304686)))))))));
+	}
+	return s1 / s2;
+}
+
+// END: rational_pcqc
+
+// BEGIN: rational_psqs
+
+/**
+* Evaluates a rational function (i.e., the ratio of two polynomials described by the coefficients stored in \\(P\\) and \\(Q\\)).
+*
+* ## Notes
+*
+* -   Coefficients should be sorted in ascending degree.
+* -   The implementation uses [Horner's rule][horners-method] for efficient computation.
+*
+* [horners-method]: https://en.wikipedia.org/wiki/Horner%27s_method
+*
+* @param x    value at which to evaluate the rational function
+* @return     evaluated rational function
+*/
+static double rational_psqs( const double x ) {
+	double ax;
+	double ix;
+	double s1;
+	double s2;
+	if ( x == 0.0 ) {
+		return -0.015625;
+	}
+	if ( x < 0.0 ) {
+		ax = -x;
+	} else {
+		ax = x;
+	}
+	if ( ax <= 1.0 ) {
+		s1 = -89.22660020080009 + (x * (-185.91953644342993 + (x * (-111.83429920482737 + (x * (-22.300261666214197 + (x * (-1.244102674583564 + (x * -0.008803330304868075)))))))));
+		s2 = 5710.502412851206 + (x * (11951.131543434614 + (x * (7264.278016921102 + (x * (1488.7231232283757 + (x * (90.59376959499312 + (x * 1.0)))))))));
+	} else {
+		ix = 1.0 / x;
+		s1 = -0.008803330304868075 + (ix * (-1.244102674583564 + (ix * (-22.300261666214197 + (ix * (-111.83429920482737 + (ix * (-185.91953644342993 + (ix * -89.22660020080009)))))))));
+		s2 = 1.0 + (ix * (90.59376959499312 + (ix * (1488.7231232283757 + (ix * (7264.278016921102 + (ix * (11951.131543434614 + (ix * 5710.502412851206)))))))));
+	}
+	return s1 / s2;
+}
+
+// END: rational_psqs
+
+/* End auto-generated functions. */
+
+/**
+* Computes the Bessel function of the first kind of order zero.
+*
+* @param x    input value
+* @return     evaluated Bessel function
+*
+* @example
+* double y = stdlib_base_j0( 0.0 );
+* // returns 1.0
+*/
+double stdlib_base_j0( const double x ) {
+	double rc;
+	double rs;
+	double y2;
+	double xc;
+	double r;
+	double y;
+	double f;
+	double s;
+	double c;
+
+	xc = x;
+	if ( xc < 0 ) {
+		xc = -xc;
+	}
+	if ( xc == STDLIB_CONSTANT_FLOAT64_PINF ) {
+		return 0.0;
+	}
+	if ( xc == 0.0 ) {
+		return 1.0;
+	}
+	if ( xc <= 4.0 ) {
+		y = xc * xc;
+		r = rational_p1q1( y );
+		f = ( xc + x1 ) * ( ( xc - ( x11 / 256.0 ) ) - x12 );
+		return f * r;
```

Comment:
The values of `x` which require change in tolerance levels gets through this block. There are only 2 such cases, for `x = 2.4044782087165135` and `x = -2.4044782087165135`. We don't have any other function being executed in this block, except `rational_p1q1`.

Reply:
Should add a comment similar to, IIRC, Spence regarding tolerance adjustment in the tests.

Reply:
We added the comment here: https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/spence/test/test.native.js#L90.
Over there, we didn't change the tolerance level in `test.js` to match with that of `test.native.js`. Should we remove the changes in `test.js` here too, and keep the increased tolerance level in `test.native.js` only, along with the comment?

Reply:
I followed what we did here: https://github.com/stdlib-js/stdlib/pull/2298#discussion_r1630783263, and yes, the same thing is happening here as well. With `-ffp-contract=off`, it works with the older tolerance levels as well.

Reply:
Nice. Thanks for reproducing! And yes, I think we can revert the tolerance change to `test.js` and just add the note to `test.native.js`.

Reply:
@kgryte Something similar happened here: https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/factorialln/test/test.native.js#L155 for `factorialln` as well. Earlier, we had the tolerance as `1.0 * EPS` in `test.js`. But, we had to change it `1.5 * EPS`, because that was required for `test.native.js`.

I checked `factorialln` similarly, with `-ffp-contract=off`. It works with `1.0`. I'll go ahead and make these changes there as well.

Reply:
Great! Thanks!